core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -415,13 +439,20 @@ class Partition(val topic: String,
           // probably unblock some follower fetch requests since log end offset has been updated
           replicaManager.tryCompleteDelayedFetch(new TopicPartitionOperationKey(this.topic, this.partitionId))
           // we may need to increment high watermark since ISR could be down to 1
-          maybeIncrementLeaderHW(leaderReplica)
+          leaderHWIncremented = maybeIncrementLeaderHW(leaderReplica)
           info
```

Comment:
You may already know this and you prefer it like it is (I won't ask you to change it), but this could also be written like:

```scala
val (info, leaderHWIncremented) = inReadLock(leaderIsrUpdateLock) {
  ...
  (info, maybeIncrementLeaderHW(leaderReplica))
}
```